lib/spack/spack/util/azure_blob.py

```diff
@@ -0,0 +1,135 @@
+# Copyright 2013-2022 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+"""This file contains the definition of the Azure Blob storage Class used to
+   integrate Azure Blob storage with spack buildcache.
+"""
+
+import datetime
+import os
+
+import llnl.util.tty as tty
+
+
+class AzureBlob:
+
+    def __init__(self, url):
+        from azure.storage.blob import BlobServiceClient
+        self.url = url
+        (self.container_name, self.blob_path) = self.get_container_blob_path()
+        if url.scheme != 'azure':
+            raise ValueError(
+                'Can not create Azure blob connection from URL with scheme: %s'
+                % (url.scheme))
+        if "AZURE_STORAGE_CONNECTION_STRING" in os.environ:
```

Comment:
Can we pass the connection string as an argument to the constructor? That will probably make the class easier to deal with in tests. We can then have a small factory function or something similar that reads the connection string from the environment and injects it into a newly created object.